File: knowledge_base/springboot/demo/src/main/java/com/kennybaik/demo/controller/UserController.java
package com.kennybaik.demo.controller;

import com.kennybaik.demo.entity.User;
import com.kennybaik.demo.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Optional;

@RestController
@RequestMapping("/users")
public class UserController {

    private final UserService userService;

    @Autowired
    public UserController(UserService userService) {
        this.userService = userService;
    }

    // 모든 사용자 조회 (GET)
    @GetMapping
    public List<User> getAllUsers() {
        return userService.getAllUsers();
    }

    // 특정 사용자 조회 (GET)
    @GetMapping("/{id}")
    public ResponseEntity<User> getUserById(@PathVariable Long id) {
        return userService.getUserById(id)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    // 사용자 추가 (POST)
    @PostMapping
    public User createUser(@RequestBody User user) {
        return userService.createUser(user);
    }

    // 사용자 삭제 (DELETE)
    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteUser(@PathVariable Long id) {
        userService.deleteUser(id);
        return ResponseEntity.noContent().build();
    }
}

File: knowledge_base/springboot/docker+mysql+aws+redis/src/main/java/com/example/demo/controller/UserController.kt
package com.example.springbootapp.controller

import com.example.springbootapp.entity.User
import com.example.springbootapp.service.UserService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/users")
class UserController(private val userService: UserService) {

    // 이메일로 사용자 조회
    @GetMapping("/{email}")
    fun getUserByEmail(@PathVariable email: String): User? {
        return userService.getUserByEmail(email)
    }

    // 모든 사용자 조회
    @GetMapping
    fun getAllUsers(): List<User> {
        return userService.getAllUsers() // 모든 사용자 조회
    }

    // 사용자 생성
    @PostMapping
    fun createUser(@RequestBody user: User): User {
        return userService.createUser(user)
    }
}
